### PR TITLE
Experience/9 23 docs refresh

### PR DESCRIPTION
**Pull request by doug-s-nava**
* updates documentation on the data fetching system to account for recent changes
* adds documentation for feature flags

Test Steps:
1. read through the docs here in a markdown format, make sure they make sense, look good, are not missing important info

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/docs/data-fetching-patterns.md [hunk 1]
```diff
@@ -49,55 +49,50 @@ Most of the work of actually making a particular request will live in hooks that
...
 
-We pass useQuery a key ([see official docs for more](https://tanstack.com/query/v4/docs/guides/query-keys)), a function which will actually do the fetch, and an options object. On each render of the custom hook, useQuery will either run, and fetch data, or not, in the case that there is already fresh data available. React-query under the hood will periodically re-run the fetch function to get fresh data. We are running with default settings for this, but this behavior can be specified as well.
+`rsUseQuery` can be used the same way that tanstack's documentation indicates, with the one caveat being that while `useQuery` is very open to multiple ways to structure the arguments passed in, `rsUseQuery` hardens the options around call signatures. `rsUseQuery` will always be called with the signature:
```
Comment (kevinhaube) [inline frontend-react/docs/data-fetching-patterns.md:56]: Did this name create any issues with React or the linter? It sounds like its use is as a hook, and they usually scream at you for not prefixing the hook name as `use`

**Review by kevinhaube — APPROVED** [review]
✅

File: frontend-react/docs/data-fetching-patterns.md (hunk 1 above)
Comment (kevinhaube) [reply frontend-react/docs/data-fetching-patterns.md:56]: If it does scream at us, `useRSQuery` could work

Comment (doug-s-nava) [reply frontend-react/docs/data-fetching-patterns.md:56]: i haven't seen any linter screaming, though in retrospect i would have expected it. My guess is that whatever logic they have to determine whether something is a hook might be confused by all of the wrapping that's going on

**Pull request merged**